File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/instruction/adapter/InsnAdapter.java
package net.yakclient.mixin.base.internal.instruction.adapter;

import net.yakclient.mixin.base.internal.instruction.Instruction;

public class InsnAdapter {
    private final InsnAdapter adapter;

    public InsnAdapter(InsnAdapter adapter) {
        this.adapter = adapter;
    }

    public InsnAdapter() {
        this.adapter = null;
    }

    public Instruction adapt(Instruction instruction) {
        if (this.adapter != null) return adapter.adapt(instruction);
        return instruction;
    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/instruction/adapter/InsnAdapter.kt
package net.yakclient.mixins.base.internal.instruction.adapter

import net.yakclient.mixins.base.internal.instruction.Instruction

open class InsnAdapter {
    private val adapter: InsnAdapter?

    constructor(adapter: InsnAdapter?) {
        this.adapter = adapter
    }

    constructor() {
        adapter = null
    }

    open fun adapt(instruction: Instruction): Instruction {
        return adapter?.adapt(instruction) ?: instruction
    }
}